compiler/ir/backend.common/src/org/jetbrains/kotlin/backend/common/lower/DefaultArgumentStubGenerator.kt

```diff
@@ -152,6 +153,11 @@ open class DefaultArgumentStubGenerator(
         newIrFunction: IrFunction,
         params: MutableList<IrValueDeclaration>
     ): IrExpression {
+        val handlerDeclaration = newIrFunction.valueParameters.last()
```

Comment:
I'd move this logic before handling of default arguments, as it's done in the old backend. Otherwise side effects of default argument intiialization can be observed before we get to assert that supercalls to methods with default arguments are not supported. This is not that important until we actually support them, but let's fix it nevertheless, otherwise we'll easily forget about this difference later.

The difference with the old backend here can be observed on the following test:
```
// TARGET_BACKEND: JVM
// WITH_RUNTIME

open class O {
    open fun foo(s: String = throw Error("Fail: this expression should not be evaluated")) {}
}

fun box(): String = try {
    val f = O::class.java.declaredMethods.single { it.name == "foo\$default" }
    f(null, O(), "s", 1, "non-null")
    "Fail: exception should have been thrown"
} catch (e: Exception) {
    val cause = e.cause
    if (cause is UnsupportedOperationException) {
        "OK"
    } else {
        cause.toString()
    }
}
```

Reply:
Done. Thanks! Reshuffled and added the test.